go/interpreter/lfvm/memory.go

```diff
@@ -188,35 +180,19 @@ func (m *Memory) SetWithCapacityAndGasCheck(offset, size uint64, value []byte, c
 	if err != nil {
 		return err
 	}
-	err = m.Set(offset, size, value)
-	if err != nil {
-		return err
-	}
-	return nil
+	return m.Set(offset, size, value)
 }
 
 func (m *Memory) SetWithCapacityCheck(offset, size uint64, value []byte) error {
 	m.EnsureCapacityWithoutGas(size)
```

Comment:
This may be a little out of scope, but I really dont understand the name of this function:
```
func (m *Memory) EnsureCapacityWithoutGas(size uint64) {
	if size <= 0 {
		return
	}
	if m.Len() < size {
		size = toValidMemorySize(size)
		fee := m.ExpansionCosts(size)
		m.total_memory_cost += fee
		m.store = append(m.store, make([]byte, size-m.Len())...)
	}
}
```
Because the changes in this PR rely on this function, could be rename/document it so that I can see if we are doing here the right thing?

Reply:
this was clarified offline